services/tables/src/main/java/com/linkedin/openhouse/tables/services/IcebergSnapshotsServiceImpl.java

```diff
@@ -65,6 +65,10 @@ public Pair<TableDto, Boolean> putIcebergSnapshots(
             icebergSnapshotRequestBody);
 
     if (tableDto.isPresent()) {
+      if (isTableLocked(tableDto.get())) {
+        authorizationUtils.checkTableLockPrivileges(
```

Comment:
The method name does not look right here? Since we are checking if an authenticated user can write to a locked table, can we move this under checkTableWritePathPrivileges and rename the method accordingly?